redbot/cogs/mutes/mutes.py

```diff
@@ -251,104 +257,124 @@ async def _handle_channel_unmutes(self):
+                        self._auto_channel_unmute_user_2(member, guild, channels)
                     )
+
                 else:
                     for channel, mute_data in channels.items():
-                        self._unmute_tasks[f"{channel}{user}"] = asyncio.create_task(
+                        task_name = f"channel-unmute-{channel}-{user}"
+                        log.debug(f"Creating task: {task_name}")
+                        if task_name in self._unmute_tasks:
+                            continue
+                        self._unmute_tasks[task_name] = asyncio.create_task(
                             self._auto_channel_unmute_user(guild.get_channel(channel), mute_data)
                         )
 
-    async def _auto_channel_unmute_user_silent(
-        self, channel: discord.abc.GuildChannel, data: dict
+    async def _auto_channel_unmute_user_2(
+        self, member: discord.Member, guild: discord.Guild, channels: Dict[int, dict]
     ):
-        """This is meant for mass channel unmute without sending the modlog case"""
-        delay = data["until"] - datetime.now(timezone.utc).timestamp()
-        if delay < 1:
-            delay = 0
-        await asyncio.sleep(delay)
-        try:
-            member = channel.guild.get_member(data["member"])
-            author = channel.guild.get_member(data["author"])
-            if not member:
-                return
-            success = await self.channel_unmute_user(
-                channel.guild, channel, author, member, _("Automatic unmute")
+
+        tasks = []
+        for channel, mute_data in channels.items():
+            author = guild.get_member(mute_data["author"])
+            tasks.append(
+                self._auto_channel_unmute_user(guild.get_channel(channel), mute_data, False)
             )
-            if success["success"]:
-                async with self.config.channel(channel).muted_users() as muted_users:
-                    if str(member.id) in muted_users:
-                        del muted_users[str(member.id)]
-            else:
-                chan_id = await self.config.guild(channel.guild).notification_channel()
-                notification_channel = channel.guild.get_channel(chan_id)
-                if not notification_channel:
-                    return
-                await notification_channel.send(
-                    _(
-                        "I am unable to mute {user} in {channel} for the following reason:\n{reason}"
-                    ).format(user=member, channel=channel.mention, reason=success["reason"])
+        result = await asyncio.gather(*tasks)
+        await modlog.create_case(
+            self.bot,
+            guild,
+            datetime.now(timezone.utc),
+            "sunmute",
+            member,
+            author,
+            _("Automatic unmute"),
+            until=None,
+        )
+        if any(result):
+            reasons = {}
+            for member, channel, reason in result:
```

Comment:
`_auto_channel_unmute_user()` can return `None` in which case this tuple-unpacking will fail.